pkg/controller/framework/controller.go

```diff
@@ -102,3 +103,117 @@ func (c *Controller) processLoop() {
 		}
 	}
 }
+
+// ResourceEventHandler can handle notifications for events that happen to a resource.
+// The events are informational only, so you can't return an error.
+// OnDelete will get the final state of the item if it is known, otherwise it will get
```

Comment:
adding description

OnUpdate could be extended to do that, but we can't guarantee that every single change will cause a distinct OnUpdate call-- sometimes changes will be aggregated (because we could have to re-list).